dds/idl/be_global.cpp

```diff
@@ -748,12 +757,108 @@ ExtensibilityKind BE_GlobalData::extensibility(AST_Decl* node) const
   return value;
 }
 
+AutoidKind BE_GlobalData::autoid(AST_Decl* node) const
+{
+  AutoidAnnotation* autoid_annotation =
+    dynamic_cast<AutoidAnnotation*>(
+      builtin_annotations_["::@autoid"]);
+  AutoidKind value = autoid_annotation->absent_value;
+  autoid_annotation->node_value_exists(node, value);
+  return value;
+}
+
+bool BE_GlobalData::id(AST_Decl* node, ACE_CDR::ULong& value) const
+{
+  IdAnnotation* id_annotation =
+    dynamic_cast<IdAnnotation*>(builtin_annotations_["::@id"]);
+  return id_annotation->node_value_exists(node, value);
+}
```

Comment:
`BE_GlobalData::get_id`?

Reply:
I need this to support hashid.

Reply:
`get_id` should be doing that, there's a todo comment for that.

Reply:
The support for hashid and others is done piece-wise in the typeobject generator.  Hence, I wanted a very specific function that just looked at `@id` and nothing else.  If we want to move that logic over because we think it is generally useful, then that's the right answer.

Reply:
Then maybe `get_id` should be made up of these more primitive functions you're adding, if I understand what you're saying correctly.

Reply:
Look at https://github.com/objectcomputing/OpenDDS/pull/1861/files#diff-67dba08556ea6d6a4cef262508eaa880R1145 and then let me know if you want to move it to `get_id`.